### PR TITLE
refactor: move all the config files to root

### PR DESCRIPTION
**Pull request by imranbarbhuiya**
**Please describe the changes this PR makes and why it should be merged:**
This pr moves all the config file to root to reduce repetition.
**Status and versioning classification:**
- This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

File: tsup.config.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+			};
+		}
+	},
+}: ConfigOptions = {}) =>
+	defineConfig({
+		clean: true,
+		dts: true,
+		entry,
+		format,
+		minify: false,
```
Comment (iCrawl) [inline tsup.config.ts:25]: This needs to be configurable. Not to mention that minify was true in a lot of configs.

File: tsup.config.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import { relative, resolve as resolveDir } from 'node:path';
```
Comment (iCrawl) [inline tsup.config.ts:1]: No need to rename this import.

File: tsup.config.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+		sourcemap,
+		target,
+		tsconfig: relative(__dirname, resolveDir(process.cwd(), 'tsconfig.json')),
+		keepNames: true,
+		globalName,
+		noExternal,
+		esbuildOptions,
+	});
+
+type ConfigOptions = Pick<
```
Comment (iCrawl) [inline tsup.config.ts:36]: Should be right after the imports.

File: packages/actions/tsup.config.ts [hunk 4]
```diff
@@ -5,4 +5,5 @@ export default createTsupConfig({
 	format: ['cjs'],
 	skipNodeModulesBundle: false,
 	noExternal: ['@actions/core'],
+	minify: true,
```
[the remark this answers is not in the document]
Comment (imranbarbhuiya) [reply packages/actions/tsup.config.ts:8]: https://github.com/discordjs/discord.js/blob/8b979c0245c42fd824d8e98745ee869f5360fc86/packages/docgen/tsup.config.ts#L9
it was false there but i'll add it now

File: packages/discord.js/.eslintrc.json [hunk 5]
```diff
@@ -19,17 +19,6 @@
...
-      2,
-      {
-        "printWidth": 120,
-        "singleQuote": true,
-        "quoteProps": "as-needed",
-        "trailingComma": "all",
-        "endOfLine": "lf",
-        "arrowParens": "avoid"
-      }
-    ],
```
Comment (iCrawl) [inline packages/discord.js/.eslintrc.json:32]: The discord.js package uses an old version of our prettier setup, can't just remove that yet sadly.

Comment (imranbarbhuiya) [reply packages/discord.js/.eslintrc.json:32]: image

This from `WebSocket.js` file

Comment (iCrawl) [reply packages/discord.js/.eslintrc.json:32]: Alright. Guess we keep it then.

**Review by iCrawl — APPROVED** [review]

File: packages/docgen/tsup.config.ts [hunk 6]
```diff
@@ -1,13 +1,5 @@
...
-	dts: true,
-	entryPoints: ['src/index.ts'],
-	format: ['esm', 'cjs'],
-	keepNames: true,
-	minify: false,
-	skipNodeModulesBundle: true,
-	sourcemap: true,
-	target: 'es2021',
+export default createTsupConfig({
+	minify: true,
```
Comment (vladfrangu) [inline packages/docgen/tsup.config.ts:4]: No point in minifying in my opinion

**Review by vladfrangu — APPROVED** [review]

**Pull request merged**